storage/postgres/storage.go

```diff
@@ -72,11 +74,24 @@ func (ps *Storage) Open(settings *storage.Settings) error {
 	ps.mutex.Lock()
 	defer ps.mutex.Unlock()
 	if ps.db == nil {
-		sslModeParam := ""
+		parsedUrl, err := url.Parse(settings.URI)
+		if err != nil {
+			return fmt.Errorf("could not parse PostgreSQL URI: %s", err)
+		}
+
+		parsedQuery, err := url.ParseQuery(parsedUrl.RawQuery)
+		if err != nil {
+			return fmt.Errorf("could not parse PostgreSQL URL query: %s", err)
+		}
+
+		parsedQuery.Add("read_timeout", strconv.Itoa(settings.ReadTimeout))
```

Comment:
what happens if the URL already contain those? i am not sure what would the URL look like in e.g. private cloud, in IBM for example it had the `sslmode=required` already part of the URL in the binding credentials.

Reply:
It will be added twice to the URL for example "postgres://postgres:postgres@localhost:5432/postgres?read_timeout=100&read_timeout=900000&sslmode=disable&write_timeout=900000".

In origin/master implementation for sslmode, you can configure the storage.uri with sslmode=required query param and  configure also storage.skip_ssl_validation=true resulting in "broken" URL "postgres://postgres:postgres@localhost:5432/postgres?sslmode=required&sslmode=disable".

So, giving priority to storage.uri query params read_timeout and write_timeout over the specific ones won't break anything simply because no one is using it yet. But giving priority to the sslmode query param can cause a regression.

Any thoughts?

Reply:
Resolved by using parsedQuery.Set(...) when setting read_timeout and write_out